core/lib/dr_app.h

```diff
@@ -130,6 +130,17 @@ dr_app_setup_and_start(void);
 DR_APP_API void
 dr_app_stop_and_cleanup(void);
 
+/**
+ * Same as dr_app_stop_and_cleanup, additionally filling in the provided
+ * dr_stats_t object, after all threads have been detached and
+ * right before clearing stats. The parameter may be NULL, case in which
+ * stats are not collected, the API behaving identically to
+ * dr_app_stop_and_cleanup().
+ */
+typedef struct _dr_stats_t dr_stats_t;
```

Comment:
Hmm, this dr_app start/stop interface is disjoint from the client interface, and currently there is no dependence here in dr_app.h on client data structures.  I thought there were users of dr_app.h who did not include dr_api.h but I'm not seeing them and dr_app.h is included inside dr_api.h these days...ruminating...

Reply:
Let me know which way to go, I'm not sure I have sufficient architectural context here to express a preference.

Reply:
Let's just stick with this struct decl as you have it.  The interfaces are already getting blurred with key use cases of having static DR and a static client linked into the app.

Reply:
Ack.